File: LT_WINDOWS/Tuần 3_LT1/0306231316_DoMinhNhat_Buoi05/Bai2/PhanSo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bai2
{
    public class PhanSo
    {
        public int tu { get; set;}
        public int mau { get; set;}

        public PhanSo() { 
            tu = 0;
            mau = 0;
        }
        public PhanSo(int tu, int mau) {
            this.tu = tu;
            this.mau = mau;
        }
        public void NhapPhanSo() {
            Console.Write("Nhap tu so: ");
            tu = int.Parse(Console.ReadLine());
            Console.Write("Nhap mau so: ");
            mau = int.Parse(Console.ReadLine());
        }
        public void XuatPhanSo() {
            Console.WriteLine("Phan so la: {0}/{1}",tu,mau);
        }
        public int UocPhanSo(int a, int b) {
            while(b != 0)
            {
                int tam = a % b;
                a = b;
                b = tam;
            }
            return a;
        }
        public void RutGonPhanSo()
        {
            int ucln = UocPhanSo(tu,mau);
            tu /= ucln;
            mau /= ucln;
        }
        public PhanSo Tong(PhanSo ps)
        {
            PhanSo sum = new PhanSo();
            sum.tu = tu * ps.mau + mau * ps.tu;
            sum.mau = mau * ps.mau;
            sum.RutGonPhanSo();
            return sum;
        }
        public PhanSo Hieu(PhanSo ps)
        {
            PhanSo sub = new PhanSo();
            sub.tu = tu * ps.mau - mau * ps.tu;
            sub.mau = mau * ps.mau;
            sub.RutGonPhanSo();
            return sub;
        }
        public PhanSo Tich(PhanSo ps)
        {
            PhanSo time = new PhanSo();
            time.tu = tu * ps.tu;
            time.mau = mau * ps.mau;
            time.RutGonPhanSo();
            return time;
        }
        public PhanSo Thuong(PhanSo ps)
        {
            PhanSo div = new PhanSo();
            div.tu = tu * ps.mau;
            div.mau = mau * ps.tu;
            return div;
        }
    }
}

File: LT_HDT/Tuần 3/Bài 1/PhanSo.cpp
#include "PhanSo.h"
#include<iostream>
#include<cmath>

using namespace std;

void PhanSo::Nhap()
{
	cout << "Nhap tu so: ";
	cin >> tu;
	cout << "Nhap mau so: ";
	cin >> mau;
}
void PhanSo::Xuat()
{
	cout << tu << "/" << mau << endl;
}
int ucln(int a, int b) {
	while (b != 0) {
		int temp = b;
		b = a % b;
		a = temp;
	}
	return a;
}
void PhanSo::RutGon() {
	int uoccln = ucln(tu, mau);
	tu /= uoccln;
	mau /= uoccln;
}
PhanSo PhanSo::Tong(PhanSo b) {
	PhanSo Sum;
	Sum.tu = tu * b.mau + mau * b.tu;
	Sum.mau = mau * b.mau;
	Sum.RutGon();
	return Sum;
}
PhanSo PhanSo::Hieu(PhanSo b) {
	PhanSo Sub;
	Sub.tu = tu * b.mau - mau * b.tu;
	Sub.mau = mau * b.mau;
	if (Sub.tu < 0)
	{
		Sub.tu = abs(Sub.tu);
		Sub.RutGon();
		Sub.tu *= (-1);
	}
	else
	{
		Sub.RutGon();
	}
	return Sub;
}
PhanSo PhanSo::Tich(PhanSo b) {
	PhanSo Time;
	Time.tu = tu * b.tu;
	Time.mau = mau * b.mau;
	Time.RutGon();
	return Time;
}
PhanSo PhanSo::Thuong(PhanSo b) {
	PhanSo Division;
	Division.tu = tu * b.mau;
	Division.mau = mau * b.tu;
	Division.RutGon();
	return Division;
}
void PhanSo::SoSanh(PhanSo b) {
	PhanSo ps = Hieu(b);
	if (ps.tu > 0) {
		cout << "Phan so thu 1 lon hon\n";
	}
	else if (ps.tu < 0)
	{
		cout << "Phan so thu 2 lon hon\n";
	}
	else
	{
		cout << "Hai phan so bang nhau";
	}
}